spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -1580,3 +1593,37 @@ CDI 2.0 introduced the following Configurators interface:
 * <<observer_method_configurator>> for `ObserverMethod` configuration
 
 The container must provide implementation for all these configurators and make them available in matching container lifecycle events as defined in <<init_events>>.
+
+[[interception_factory]]
+
+=== Apply interceptor programmatically
+
+CDI 2.0 introduces the `javax.enterprise.inject.spi.InterceptionFactory<T>` interface to add services to an instance not created by the container by making each of its method invocation, a business method invocation as defined in <<biz_method>>.
+It will bind business methods only to `@AroundInvoke` interceptor methods.
+
+
+[source, java]
+----
+public interface InterceptionFactory<T> {
+    InterceptionProxyFactory<T> ignoreFinalMethods();
+    AnnotatedTypeConfigurator<T> configure();
+    T createInterceptedInstance(T instance);
+}
+----
+
+* `ignoreFinalMethods()` `ignoreFinalMethods()` Instructs the container to ignore all non-static, final methods with public, protected or default visibility declared on any bean type of the specific bean during validation of injection points that require proxyable bean type.
+These method should never be invoked upon bean instances. Otherwise, unpredictable behavior results.
+ It will bypass standard rules defined in <<unproxyable>>.
+* `configure()` returns an `AnnotatedTypeConfigurator` (as defined in <<annotated_type_configurator>>) initialized with the instance type to easily apply specific interceptor binding to apply when enhancing the instance.
+The method always return the same `AnnotatedTypeConfigurator`
+* `createInterceptedInstance()` returns an enhanced version of the instance for which each method invocations will be a business method invocation.
+The method can be only called once, subsequent calls will throw an exception.
```

Comment:
Sorry I didn't have enough time to go through this PR. This sentence is too general.